### PR TITLE
Add a user context

### PR DESCRIPTION
**Pull request by ciradu2204**
solves #20 
- [x] Added a user context file under the src folder
- [x] Created an interface to define the user object types 
- [x] Created a UserContextType to define the types for both the user and updateUser function
- [x] Initialized the user context 
- [x] Created a user provider and exported it
- [x] Created a useUser hook to access both the user and the updateUser function
- [x] Imported the useUser hook in the index.tsx

**How to test**
- Click the login button
- View the login user name displayed

### CODE REVIEW

File: packages/frontend/src/UserContext.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+export type USERContextType = {
+    user?:USER,
+    updateUser: () => void
+}
+
+//initialized as null
+const UserContext = createContext<USERContextType>({
+    user:undefined,
```
Comment (pavankpdev) [inline packages/frontend/src/UserContext.tsx:17]: Please do not init as `undefined` make it an empty object.

File: packages/frontend/src/UserContext.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+interface USER{
+     fullName: string,
+     email: string,
+     walletAddress: string,
+     avatar: string
+}
+
+export type USERContextType = {
+    user?:USER,
```
Comment (pavankpdev) [inline packages/frontend/src/UserContext.tsx:11]: Marking user as defined is not a valid, instead use a union 
`user: IUser | Record<string, string>`

File: packages/frontend/src/UserContext.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React, { createContext, useContext, useState } from "react";
+
+interface USER{
```
Comment (pavankpdev) [inline packages/frontend/src/UserContext.tsx:3]: It is a best Practice to prefix the `interface` name with `I` example rename this interface name to `IUser`

File: packages/frontend/src/UserContext.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+const UserContext = createContext<USERContextType>({
+    user:undefined,
+    updateUser: () => {}
+});
+
+export const UserProvider = ({children}:any) => {
+    const [user, setUser] =  useState<USER| undefined>(undefined)
+
+    const updateUser = () =>{
+       //retrieve user details from db/local storage
```
Comment (pavankpdev) [inline packages/frontend/src/UserContext.tsx:25]: Mark this as TODO, since this should be done later

File: packages/frontend/src/pages/_app.tsx [hunk 5]
```diff
@@ -3,11 +3,14 @@ import type {AppProps} from "next/app";
 
 // THEME
 import ArtPlaceTheme from "@/theme";
+import { UserProvider } from '../UserContext';
```
Comment (pavankpdev) [inline packages/frontend/src/pages/_app.tsx:6]: Use import alias, such as `@/context/User.tsx`

File: packages/frontend/src/pages/index.tsx [hunk 6]
```diff
@@ -1,9 +1,12 @@
-import {Heading} from "@chakra-ui/react";
-
+import {Heading, Button} from "@chakra-ui/react";
+import {useUser }from "../UserContext"
 export default function Home() {
+  const {user, updateUser} = useUser()
   return (
     <>
         <Heading>ArtPlace</Heading>
+        {!user && <button onClick={updateUser}>Click to Login</button>}
```
Comment (pavankpdev) [inline packages/frontend/src/pages/index.tsx:8]: This is not necessary as we'll get a auth hook from rainbow kit

File: packages/frontend/src/UserContext.tsx
Comment (pavankpdev) [inline packages/frontend/src/UserContext.tsx:1]: Please move this file to following directory. 
`context/User.tsx`

**Review by pavankpdev — CHANGES_REQUESTED** [review]

File: packages/frontend/src/UserContext.tsx (hunk 4 above)
Comment (ciradu2204) [reply packages/frontend/src/UserContext.tsx:25]: so I should update the comments and remove the implementation?

**Comment by ciradu2204** [comment]
@pavankpdev, could I add the yarn.lock in the .gitignore file?

**Comment by pavankpdev** [comment]
> @pavankpdev, can I add the yarn.lock in the .gitignore file?

no please don't add that, having lock file would help us all stay in sync with package versions

**Comment by pavankpdev** [comment]
let me know if all the changes are done and ready to be reviews @ciradu2204

**Comment by ciradu2204** [comment]
Yes all the changes are done

**Review by pavankpdev — APPROVED** [review]
Great Merging

**Comment by pavankpdev** [comment]
@ciradu2204 can you share your discord username, and also join our server if you're not in.
[https://discord.gg/cXKMsx5h](https://discord.gg/cXKMsx5h)

**Pull request merged**